modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -193,11 +193,11 @@ class ReplCalculator(
         if (queue.nonEmpty) {
           //queue could be empty if InterruptRequest was asked!
           log.debug("Dequeuing execute request current size: " + queue.size)
-          queue = queue.dequeue._2
-          queue.headOption foreach { case (ref, er) =>
-            log.debug("About to execute request from the queue")
-            execute(ref, er)
-          }
+          val (executeRequest, newQueue) = queue.dequeue
+          queue = newQueue
+          val (ref, er) = executeRequest
```

Comment:
> Here you'll actually always re-execute the first entry if I'm not wrong.

I dont think so. Here's doc for `deque`:
```scala
/** Returns a tuple with the first element in the queue,
   *  and a new queue with this element removed.
   *
   *  @throws Predef.NoSuchElementException
   */
  def dequeue: (A, Queue[A]) =  ...
```

also if I'm not mistaken, in the old code, if there were always were more than 2 items in the queue, `deque`ing before `queue.headOption` would skip (not execute!) each second item.

This is why, case when queue is empty is changed into simply:
```scala
queue.enqueue
self ! "process-next"
```